docs/modules/ROOT/pages/cops_rspec.adoc

```diff
@@ -2710,6 +2710,26 @@ end
 it 'validates users' do
   expect([user1, user2, user3]).to all(be_valid)
 end
+
+# bad
+it 'sets inferred fields for users' do
+  [user1, user2, user3].each do |user|
+    expect(user).to have_attributes(
+      post_count: PostService.new(user).post_count,
+      karma: KarmaService.new(user).value
+    )
+  end
+end
+
+# good
+it 'sets inferred fields for users' do
+  expect([user1, user2, user3]).to all(satisfy do |user|
```

Comment:
Would the following work?
‘’’
expect([user1, user2, user3]).to all have_attributes(
‘’’

Reply:
That would certainly be more readable, but unfortunately not in this scenario (where block arguments are used). The challenge is that the `have_attributes` call is receiving the `user` as an argument. It would raise "undefined local variable or method `user'"